File: src/SampleCRM/Helpers/CountryCodeToCountryNameConverter.cs
using System;
using System.Globalization;
using System.Linq;
using System.Windows.Data;

namespace SampleCRM.Web.Views
{
    public class CountryCodeToCountryNameConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (value != null)
            {
                var countryCode = value as string;
                if (!string.IsNullOrEmpty(countryCode))
                {
                    var countryCodesContext = new CountryCodesContext();
                    var query = countryCodesContext.GetCountryByIdQuery(countryCode);
                    var entityObj = countryCodesContext.Load(query);
                    value = entityObj.Entities.FirstOrDefault();
                }
            }

            return value;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            return value;
        }
    }
}

File: src/SampleCRM/Helpers/CountryCodeToCountryNameConverter.vb
Imports System.Globalization
Imports System.Windows.Data
Imports SampleCRM.Web.SampleCRM.Web

Namespace Views
    Public Class CountryCodeToCountryNameConverter
        Implements IValueConverter

        Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
            If value IsNot Nothing Then
                Dim countryCode As String = TryCast(value, String)
                If Not String.IsNullOrEmpty(countryCode) Then
                    Dim countryCodesContext As New CountryCodesContext()
                    Dim query = countryCodesContext.GetCountryByIdQuery(countryCode)
                    Dim entityObj = countryCodesContext.Load(query)
                    value = entityObj.Entities.FirstOrDefault()
                End If
            End If

            Return value
        End Function

        Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Return value
        End Function
    End Class
End Namespace
